shaal/ddev
# Add PHP Debug + Shell Debug to VSCode

## The Problem/Issue/Bug:
When using Gitpod, it's not possible to debug shell scripts, and it's not possible to test `ddev xdebug`.

## How this PR Solves The Problem:
Adding the ability to run PHP Debug and Shell Debug in VSCode

## Manual Testing Instructions:
### In Gitpod:
Test Bash debug:
1. Click on Debug panel, choose Bash, and then choose a script to debug.
1. Confirm that you can run the script step-by-step.

Test PHP Debug:
1. Run in Terminal: `code .`
1. (It will open new tab with VSCode)
1. Run in Terminal: `ddev xdebug`
1. Click on Debug panel, choose `Listen to Xdebug`
1. Create a breakpoint in `web/index.php`
1. Run in Terminal: `gp preview $(gp url 8080)`
1. Confirm that the debugger stopped at the breakpoint.

## Automated Testing Overview:


## Related Issue Link(s):

## Release/Deployment notes:




<a href="https://gitpod.io/#https://github.com/shaal/ddev/pull/4"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
